tests/test_mapper_representative_time_to_datetime.py

```diff
@@ -0,0 +1,139 @@
+import pandas as pd
+from sqlalchemy import Engine, MetaData
+from zoneinfo import ZoneInfo
+import pytest
+from datetime import datetime, timedelta
+
+from chronify.sqlalchemy.functions import read_database, write_database
+from chronify.time_series_mapper import map_time
+from chronify.time_configs import DatetimeRange
+from chronify.models import TableSchema
+from chronify.time import TimeIntervalType, MeasurementType
+from chronify.exceptions import ConflictingInputsError, MissingParameter
+from chronify.datetime_range_generator import DatetimeRangeGenerator
+
+
+def generate_datetime_data(time_config: DatetimeRange) -> pd.Series:
+    return pd.to_datetime(list(DatetimeRangeGenerator(time_config).iter_timestamps()))
+
+
+def get_datetime_schema(year, tzinfo) -> TableSchema:
+    start = datetime(year=year, month=1, day=1, tzinfo=tzinfo)
+    end = datetime(year=year + 1, month=1, day=1, tzinfo=tzinfo)
+    resolution = timedelta(hours=1)
+    length = (end - start) / resolution + 1
+    schema = TableSchema(
+        name="mapped_data",
+        time_config=DatetimeRange(
+            start=start,
+            resolution=resolution,
+            length=length,
+            interval_type=TimeIntervalType.PERIOD_BEGINNING,
+            time_column="timestamp",
+        ),
+        time_array_id_columns=["id"],
+        value_column="value",
+    )
+    return schema
+
+
+def run_test(
+    engine: Engine,
+    df: pd.DataFrame,
+    from_schema: TableSchema,
+    to_schema: TableSchema,
+    error: tuple[any, str],
+) -> None:
+    # Ingest
+    metadata = MetaData()
+    with engine.connect() as conn:
+        write_database(df, conn, from_schema.name, from_schema.time_config)
+        conn.commit()
+    metadata.reflect(engine, views=True)
+
+    # Map
+    if error:
+        with pytest.raises(error[0], match=error[1]):
+            map_time(engine, metadata, from_schema, to_schema)
+    else:
+        map_time(engine, metadata, from_schema, to_schema)
+
+        # Check mapped table
+        with engine.connect() as conn:
+            query = f"select * from {to_schema.name}"
+            queried = read_database(query, conn, to_schema.time_config)
+        queried = queried.sort_values(by=["id", "timestamp"]).reset_index(drop=True)
+
+        truth = generate_datetime_data(to_schema.time_config)
+        check_mapped_table(queried, truth)
+
+
+def check_mapped_table(df: pd.DataFrame, ts: pd.Series) -> None:
+    res = sorted(df["timestamp"].drop_duplicates().tolist())
```

Comment:
In other places you call `Series.to_list()`. In some previous comment I was asking for consistency. I do think both versions are available, but I wouldn't be surprised if they eventually remove this one.